public/components/reindex.react.js

```diff
@@ -52,18 +62,30 @@ export default class ReindexComponent extends React.Component {
                         </Col>
                         <Col xs={6} md={6}>
                             <Panel header="Details">
-                                <ContentSource key={this.state.contentSource.id}
-                                               id={this.state.contentSource.id}
-                                               description={this.state.contentSource.description}
-                                               appName={this.state.contentSource.appName}
-                                               reindexEndpoint={this.state.contentSource.reindexEndpoint}/>
+                                {this.state.editModeOn ?
+                                    <ContentSourceEdit key={this.state.contentSource.id}
```

Comment:
If you're worried about a contentSource object being incomplete maybe look into PropTypes which will warn if you said data into a component that it's not expecting.